lib/stdlib/src/erl_error.erl

```diff
@@ -536,10 +536,20 @@ location(L) ->
 sep(1, S) -> S;
 sep(_, S) -> [$\n | S].
 
+is_rec_init(F) when is_atom(F) ->
+    case atom_to_binary(F) of
+        <<"rec_init$^", _/binary>> -> true;
+        _ -> false
+    end;
+is_rec_init(_) -> false.
+
 origin(1, M, F, A) ->
     case is_op({M, F}, n_args(A)) of
         {yes, F} -> <<"in operator ">>;
-        no -> <<"in function ">>
+        no -> case is_rec_init(F) of
+            true -> <<"in record">>;
+            _ -> <<"in function ">>
+        end
```

Comment:
I recommend not using `_` to match a single known value. The `_` can hide bugs if `is_rec_init/1` is changed in the future to return more values. Also, if `true` is misspelled as `ture`, that bug is hidden too. Furthermore, the '_' can result in worse types in both Dialyzer and the compiler. Sometimes those worse types can result in worse code because the compiler is unable to see that some optimization is safe.

In this particular case, the compiler will "see" that if the value is not `true`, it must be false. In other words, the compiler will do the optimization for you.

```suggestion
        no ->
            case is_rec_init(F) of
                true -> <<"in record">>;
                false -> <<"in function ">>
            end
```